querycheck/expect_known_value_test.go

```diff
@@ -0,0 +1,143 @@
+				// ```provider "examplecloud" {}``` has a slightly different syntax for a .tfquery.hcl file
+				// provider bock simulates a real providers workflow
+				// "config" in this case means configuration of the list resource/filters
+
+				Query: true,
+				Config: `
+				provider "examplecloud" {} 
+
+				list "examplecloud_containerette" "test" {
+					provider         = examplecloud
+					include_resource = true
+
+					config {
+						resource_group_name = "foo"
+ 					}
+				}
+				`,
+				QueryResultChecks: []querycheck.QueryResultCheck{
+					querycheck.ExpectKnownValue(
+						"examplecloud_containerette.test",
+						"banane",
+						tfjsonpath.New("instances"),
+						knownvalue.NumberExact(big.NewFloat(5)),
+					),
+				},
+			},
+		},
+	})
+}
+
+// Let's add a test case that checks the failure scenario when the value is incorrect.
+func TestExpectKnownValue_ValueIncorrect(t *testing.T) {
+	t.Parallel()
+
+	r.UnitTest(t, r.TestCase{
+		TerraformVersionChecks: []tfversion.TerraformVersionCheck{
+			tfversion.SkipBelow(tfversion.Version1_14_0),
+		},
+		ProtoV6ProviderFactories: map[string]func() (tfprotov6.ProviderServer, error){
+			"examplecloud": providerserver.NewProviderServer(testprovider.Provider{
+				ListResources: map[string]testprovider.ListResource{
+					"examplecloud_containerette": examplecloudListResource(),
+				},
+				Resources: map[string]testprovider.Resource{
+					"examplecloud_containerette": examplecloudResource(),
+				},
+			}),
+		},
+		Steps: []r.TestStep{
+			{ // config mode step 1 needs tf file with terraform providers block
+				// this step should provision all the resources that the query is support to list
+				// for simplicity we're only "provisioning" one here
+				Config: `
+				resource "examplecloud_containerette" "primary" {
+					name                = "banana"
+					resource_group_name = "foo"
+					location  			= "westeurope"
+			
+					instances = 5
+				}`,
+			},
+			{ // Query mode step 2, operates on .tfquery.hcl files (needs tf file with terraform providers block)
+				// ```provider "examplecloud" {}``` has a slightly different syntax for a .tfquery.hcl file
+				// provider bock simulates a real providers workflow
+				// "config" in this case means configuration of the list resource/filters
+
+				Query: true,
+				Config: `
+				provider "examplecloud" {} 
+
+				list "examplecloud_containerette" "test" {
+					provider         = examplecloud
+					include_resource = true
+
+					config {
+						resource_group_name = "foo"
+ 					}
+				}
+				`,
+				QueryResultChecks: []querycheck.QueryResultCheck{
+					querycheck.ExpectKnownValue(
+						"examplecloud_containerette.test",
+						"banana",
+						tfjsonpath.New("instances"),
+						knownvalue.NumberExact(big.NewFloat(4)),
+					),
+				},
+				ExpectError: regexp.MustCompile("examplecloud_containerette.test - the resource banana was not found"),
```

Comment:
This test should validate that the `ExpectKnownValue` check correctly flags values that are incorrect, but here we're checking for the presence of the resource.